library/Icinga/Application/Config.php

```diff
@@ -412,17 +412,16 @@ public static function app($configname = 'config', $fromDisk = false)
      */
     public static function module($modulename, $configname = 'config', $fromDisk = false)
     {
-        if (! isset(self::$modules[$modulename])) {
-            self::$modules[$modulename] = array();
+        if (! isset(static::$modules[$modulename])) {
+            static::$modules[$modulename] = array();
```

Comment:
I absolutely second this when it goes to method calls. Disagree for properties, not because it is wrong, but because it gives a developer extending your class the wrong feeling about having the possibility to work with his own $property, while instead sharing it with all other classes inheriting from the same base - unless they explicitly override the property by defining their own one. Instead when reading 'self' in the base class, he will hopefully think twice about what would be going on. Not saying that there aren't cases where static might absolutely be justified. Just, they break when the first one writes a class not re-declaring the very same property. So, my concerns here are about developers hurting themselves. @lippserd?